client/allocrunner/taskrunner/task_runner_hooks.go

```diff
@@ -89,26 +89,31 @@ func (tr *TaskRunner) initHooks() {
 		}))
 	}
 
+	// Get the consul namespace for the TG of the allocation
+	consulNamespace := tr.alloc.ConsulNamespace()
+
 	// If there are templates is enabled, add the hook
 	if len(task.Templates) != 0 {
 		tr.runnerHooks = append(tr.runnerHooks, newTemplateHook(&templateHookConfig{
-			logger:       hookLogger,
-			lifecycle:    tr,
-			events:       tr,
-			templates:    task.Templates,
-			clientConfig: tr.clientConfig,
-			envBuilder:   tr.envBuilder,
+			logger:          hookLogger,
+			lifecycle:       tr,
+			events:          tr,
+			templates:       task.Templates,
+			clientConfig:    tr.clientConfig,
+			envBuilder:      tr.envBuilder,
+			consulNamespace: consulNamespace,
```

Comment:
For the service hooks we don't need to do anything; the agent's consul namespace is [plumbed](https://github.com/hashicorp/nomad/blob/5c3399853d297d4af0a21ae7309f4e7b2267f99d/nomad/structs/config/consul.go) into the `consul/api.Client` on startup. If the namespace is configured in the job, that gets set in the service definition registration and takes precedence over the one in the otherwise default `consul/api.Client`.